apps/server/src/shared/repo/school/school.repo.ts

```diff
@@ -24,9 +24,12 @@ export class SchoolRepo extends BaseDORepo<SchoolDO, School, ISchoolProperties>
 	}
 
 	async findBySchoolNumber(officialSchoolNumber: string): Promise<SchoolDO | null> {
-		const school: School | null = await this._em.findOne(School, { officialSchoolNumber });
+		const [schools, count] = await this._em.findAndCount(School, { officialSchoolNumber });
```

Comment:
Sound like you want em.findOneOrFail(School, { officialSchoolNumber });

Reply:
We considered using findOne or fail, but decided against it because we only want to throw the InternalServerError in that specific case and not on every fail